### PR TITLE
Bump deploy k8s versions

### PR DESCRIPTION
**Pull request by mowangdk**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup
**What this PR does / why we need it**:
Bump up k8s version of deploy yaml

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/120997

**Special notes for your reviewer**:
None

**Does this PR introduce a user-facing change?**:

```release-note
Update deploy directory structure and update sidecar releases, corresponding to Kubernetes 1.28.
```

### CODE REVIEW

**Comment by mowangdk** [comment]
@xing-yang ptal~

**Comment by xing-yang** [comment]
Can you update the sidecar images to the latest?
Can you also add a release note?

**Comment by mowangdk** [comment]
> Can you update the sidecar images to the latest? Can you also add a release note?

I was going to make another pr to upgrade  sidecar images, but of course it's ok to upgrade in this pr~

File: deploy/kubernetes-1.28-test [hunk 1]
```diff
@@ -0,0 +1 @@
+kubernetes-1.27-test
```
Comment (xing-yang) [inline deploy/kubernetes-1.28-test:1]: We don't need to add kubernetes-1.28 and kubernetes-1.28-test yet.

Comment (mowangdk) [reply deploy/kubernetes-1.28-test:1]: okay~

**Comment by mowangdk** [comment]
> Can you also add a release note?

Should I put this release note in the pr description? Or create a CHANGELOG-1.12.md without releasing a new version?

**Comment by xing-yang** [comment]
In the PR description, there is this question "Does this PR introduce a user-facing change?:", please add release note there inside the "release note" section.

**Comment by xing-yang** [comment]
I added release note so this can be merged.

/lgtm
/approve

**Comment by mowangdk** [comment]
> In the PR description, there is this question "Does this PR introduce a user-facing change?:", please add release note there inside the "release note" section. This way, this note will be added automatically to the changelog later.


got 👍

**Pull request merged**